File: app/src/main/java/com/example/david/lists/widget/configactivity/buildlogic/WidgetConfigComponent.java
package com.example.david.lists.widget.configactivity.buildlogic;

import android.app.Application;

import androidx.appcompat.app.AppCompatActivity;

import com.example.david.lists.common.buildlogic.ViewCommonModule;
import com.example.david.lists.common.buildlogic.ViewScope;
import com.example.david.lists.widget.configactivity.WidgetConfigActivity;

import dagger.BindsInstance;
import dagger.Component;

@ViewScope
@Component(modules = {
        WidgetConfigModule.class,
        ViewCommonModule.class
})
public interface WidgetConfigComponent {
    void inject(WidgetConfigActivity activity);

    @Component.Builder
    interface Builder {
        WidgetConfigComponent build();

        @BindsInstance
        Builder application(Application application);

        @BindsInstance
        Builder activity(AppCompatActivity appCompatActivity);

        @BindsInstance
        Builder widgetId(int widgetId);
    }
}

File: app/src/main/java/com/precopia/david/lists/widget/configview/buildlogic/WidgetConfigComponent.kt
package com.precopia.david.lists.widget.configview.buildlogic

import android.app.Application
import android.content.Context
import androidx.appcompat.app.AppCompatActivity
import com.precopia.david.lists.common.buildlogic.ViewCommonModule
import com.precopia.david.lists.common.buildlogic.ViewScope
import com.precopia.david.lists.widget.common.buildlogic.WidgetCommonModule
import com.precopia.david.lists.widget.configview.WidgetConfigView
import dagger.BindsInstance
import dagger.Component

@ViewScope
@Component(modules = [
    WidgetConfigModule::class,
    WidgetCommonModule::class,
    ViewCommonModule::class
])
interface WidgetConfigComponent {

    fun inject(view: WidgetConfigView)

    @Component.Builder
    interface Builder {
        fun build(): WidgetConfigComponent

        @BindsInstance
        fun view(view: AppCompatActivity): Builder

        @BindsInstance
        fun application(application: Application): Builder

        @BindsInstance
        fun context(context: Context): Builder
    }
}
